crates/sui-adapter/src/execution_engine.rs

```diff
@@ -325,6 +325,29 @@ fn execution_loop<
                     tx_ctx,
                 )?;
             }
+            SingleTransactionKind::Genesis(GenesisTransaction { objects }) => {
+                if tx_ctx.epoch() != 0 {
+                    panic!("BUG: Genesis Transactions can only be executed in epoch 0");
+                }
+
+                for genesis_object in objects {
+                    match genesis_object {
+                        sui_types::messages::GenesisObject::RawObject { data, owner } => {
+                            let object = Object {
+                                data,
+                                owner,
+                                previous_transaction: tx_ctx.digest(),
+                                storage_rebate: 0,
```

Comment:
This seems like the right thing to do here, but just noting that this will lead to some slightly odd behavior the first time a genesis object is used. E.g., if a we have a genesis coin that is `N` bytes, we'll end up paying `N` * `storage_cost` the first time we send a tx that uses the coin, and the `storage_rebate` for the coin will be `N * storage_cost` afterward.

Reply:
I wonder if we should actually do the accounting properly by initializing the storage fund pool to be the total storage cost of all the genesis objects, and then put the correct storage_rebate in the genesis objects?

Reply:
So this is a problem today with the current way genesis creates these objects, it just becomes a bit more apparent now. I think that we're still a ways away from genesis being in a state we want it in and I do think that this is something we could consider fixing.